src/main/java/org/wso2/carbon/connector/KafkaProduceConnector.java

```diff
@@ -584,6 +588,130 @@ private Object handleComplexAvroTypes(Object jsonString, Schema schema) {
+                        "for input: " + input + " The input needs to be in the yyyy-MM-dd'T'HH:mm:ss.SSSSSS'Z' format.");
+            }
+        }
+        if (LogicalTypes.localTimestampMillis().equals(schema.getLogicalType())) {
+            try {
+                String timestampString = preprocessTimeString(input,
+                        KafkaConnectConstants.REGEX_FOR_MILLIS_PART_WITHOUT_TIME_ZONE, "3");
+                Date date = DateUtils.parseDate(timestampString, "yyyy-MM-dd'T'HH:mm:ss.SSS");
+                return Utils.toEpochMillis(date);
+            } catch (ParseException e) {
+                throw new SerializationException("Error serializing Avro message of type long with logicalType date " +
+                        "for input: " + input + " The input needs to be in the 'yyyy-MM-dd'T'HH:mm:ss.SSS' format.");
+            }
+        }
+        if (LogicalTypes.localTimestampMicros().equals(schema.getLogicalType())) {
+            try {
+                String timestampString = preprocessTimeString(input,
+                        KafkaConnectConstants.REGEX_FOR_MICROS_PART_WITHOUT_TIME_ZONE, "6");;
+                Date date = DateUtils.parseDate(timestampString, "yyyy-MM-dd'T'HH:mm:ss.SSSSSS");
+                return Utils.toEpochMicros(new Timestamp(date.getTime()));
+            } catch (ParseException e) {
+                throw new SerializationException("Error serializing Avro message of type long with logicalType date " +
+                        "for input: " + input + " The input needs to be in the 'yyyy-MM-dd'T'HH:mm:ss.SSSSSS' format.");
+            }
+        }
+        try {
+            return Long.valueOf(input);
+        } catch (NumberFormatException e) {
+            throw new SerializationException(
+                    "Error serializing Avro message of type long for input: " + input, e);
+        }
+    }
+
+    /**
+     * Preprocesses a time string to ensure that the milliseconds or microseconds part has exactly
+     * the specified number of digits.
+     * <p>
+     * This method searches for the milliseconds or microseconds part in the given time string using
+     * the provided regular expression. If the part is found, it pads it with zeros to ensure it has exactly
+     * the specified number of digits.
+     *
+     * @param timeString the original time string to be processed.
+     * @param regex the regular expression to find the milliseconds or microseconds part in the time string.
+     * @param noOfDigit the number of digits that the milliseconds or microseconds part should have after processing.
+     * @return the processed time string with the milliseconds or microseconds part formatted to the specified
+     * number of digits, or the original time string if the part is not found.
+     */
+    private static String preprocessTimeString(String timeString, String regex, String noOfDigit) {
+        Pattern pattern = Pattern.compile(regex);
```

Comment:
Let's have the patterns compiled as class level variables.

Reply:
Fixed